src/runtime/auto-prompt-manager.js

```diff
@@ -365,56 +376,62 @@ export class AutoPromptManager {
    *   dismissedPrompts: (?string|undefined),
    *   shouldShowAutoPrompt: (boolean|undefined),
    * }} params
-   * @return {!string|undefined}
+   * @return {!Promise<string|undefined>}
    */
   getAudienceActionPromptType_({
     article,
     autoPromptType,
     dismissedPrompts,
     shouldShowAutoPrompt,
   }) {
-    let potentialActions = article?.audienceActions?.actions || [];
-    potentialActions = potentialActions.filter((action) =>
-      this.checkActionEligibility_(action.type)
-    );
-
-    // No audience actions means use the default prompt.
-    if (potentialActions.length === 0) {
-      return undefined;
-    }
-
-    // Default to the first recommended action.
-    let actionToUse = potentialActions[0].type;
-
-    // Contribution prompts should appear before recommended actions, so we'll need
-    // to check if we have shown it before.
-    if (
-      autoPromptType === AutoPromptType.CONTRIBUTION ||
-      autoPromptType === AutoPromptType.CONTRIBUTION_LARGE
-    ) {
-      if (!dismissedPrompts) {
-        this.promptDisplayed_ = AutoPromptType.CONTRIBUTION;
-        return undefined;
-      }
-      const previousPrompts = dismissedPrompts.split(',');
-      potentialActions = potentialActions.filter(
-        (action) => !previousPrompts.includes(action.type)
+    const audienceActions = article?.audienceActions?.actions || [];
+
+    return this.getEvent_(
+      COMPLETED_ACTION_TO_STORAGE_KEY_MAP.get(
+        AnalyticsEvent.ACTION_SURVEY_DATA_TRANSFER
+      )
+    ).then((surveysCompleted) => {
+      // No audience actions means use the default prompt.
+      let potentialActions = audienceActions.filter((action) =>
+        this.checkActionEligibility_(action.type, surveysCompleted)
       );
 
-      // If all actions have been dismissed or the frequency indicates that we
-      // should show the Contribution prompt again regardless of previous dismissals,
-      // we don't want to record the Contribution dismissal
-      if (potentialActions.length === 0 || shouldShowAutoPrompt) {
+      if (potentialActions.length === 0) {
```

Comment:
Could you add a comment for this check?

Reply:
Maybe that comment from line 394 could move down so it's just above this check, like before?

Reply:
Ah yes, this was unintended. Thanks for pointing this out.